File: DataStructureAndAlgorithm/Searching/BFS/Python/bfs.py
from collections import deque

def bfs(graph, start_node):
    visited = set()
    queue = deque([start_node])
    visited.add(start_node)

    while queue:
        node = queue.popleft()
        print(f"Visited {node}")

        for neighbor in graph.get(node, []):
            if neighbor not in visited:
                visited.add(neighbor)
                queue.append(neighbor)

File: DataStructureAndAlgorithm/Searching/BFS/C++/BFS.cpp
#include "BFS.h"
#include <iostream>
#include <queue>

void BFS(int startNode, int numNodes, const std::vector<std::list<int>>& adj) {
    std::vector<bool> visited(numNodes, false);
    std::queue<int> q;

    visited[startNode] = true;
    q.push(startNode);

    while (!q.empty()) {
        int currentNode = q.front();
        q.pop();
        std::cout << "Visited " << currentNode << std::endl;

        for (int neighbor : adj[currentNode]) {
            if (!visited[neighbor]) {
                visited[neighbor] = true;
                q.push(neighbor);
            }
        }
    }
}
